agent/engine/dockerclient/versions.go

```diff
@@ -30,6 +30,7 @@ const (
 	Version_1_20 DockerVersion = "1.20"
 	Version_1_21 DockerVersion = "1.21"
 	Version_1_22 DockerVersion = "1.22"
+	Version_1_23 DockerVersion = "1.23"
```

Comment:
Verified manually by comparing fields of https://docs.docker.com/engine/reference/api/docker_remote_api_v1.23/#/create-a-container with https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#/create-a-container. Also created tracking issue to handle changes related to HostConfig in Docker 1.24 https://github.com/aws/amazon-ecs-agent/issues/529